### PR TITLE
refactor(indexer): Attach local receipt to chunk. Remove local_receipts field from StreamerMessage

### LINKED ISSUE #3497: refactor(indexer): Attach local receipt to chunk
Currently local receipts not attached to specific chunk. But it should be attached to the same chunk as parent transaction.

### PR DESCRIPTION
**Pull request by khorolets**
Closes #3497

### CODE REVIEW

File: chain/indexer/src/streamer/streamer.rs [hunk 1]
```diff
@@ -226,11 +222,16 @@ async fn build_streamer_message(
             .await?,
         );
 
+        // Prepending local receipts with common receipts from current chunk
+        // local ones will be required below to attach them to corresponding execution outcomes
+        let mut chunk_receipts = local_receipts.clone();
```
Comment (frol) [inline chain/indexer/src/streamer/streamer.rs:227]: You don't need to accumulate and clone the local receipts. Just prepend the result of `convert_transactions_sir_into_local_receipts` to the non-local receipts (let's rename the field while we unpacking:

```rust
let views::ChunkView { transactions, author, header, receipts: non_local_receipts } = chunk;

let local_receipts = convert_transactions_sir_into_local_receipts(...).await?;
let mut receipts = local_receipts;
receipts.extend(non_local_receipts);
```

**Review by frol — CHANGES_REQUESTED** [review]

**Comment by khorolets** [comment]
@frol please have a look at clippy warning

```
warning: module has the same name as its containing module
 --> chain/indexer/src/streamer/mod.rs:7:1
  |
7 | mod streamer;
  | ^^^^^^^^^^^^^
  |
  = note: `#[warn(clippy::module_inception)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#module_inception
```

I guess we should rename the `streamer/streamer.rs`. I can suggest 
* stream
* stream_message
* handler
* streamer_message

File: chain/indexer/src/streamer/fetchers.rs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+pub struct IndexerTransactionWithOutcome {
+    pub transaction: views::SignedTransactionView,
+    pub outcome: IndexerExecutionOutcomeWithReceipt,
+}
+
+#[derive(Clone, Debug)]
+pub struct IndexerExecutionOutcomeWithReceipt {
+    pub execution_outcome: views::ExecutionOutcomeWithIdView,
+    pub receipt: Option<views::ReceiptView>,
+}
```
Comment (frol) [inline chain/indexer/src/streamer/fetchers.rs:44]: Let's have them in `types.rs`

**Review by frol — APPROVED** [review]
Feel free to merge without my further review.

File: chain/indexer/src/streamer/fetchers.rs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+use actix::Addr;
+use futures::stream::StreamExt;
+use tracing::warn;
+
+pub use near_primitives::hash::CryptoHash;
+pub use near_primitives::{types, views};
+
+use self::super::types::{ExecutionOutcomesWithReceipts, IndexerExecutionOutcomeWithReceipt};
+use super::INDEXER;
+use crate::streamer::errors::FailedToFetchData;
```
Comment (frol) [inline chain/indexer/src/streamer/fetchers.rs:12]:
```suggestion
use super::INDEXER;
use super::errors::FailedToFetchData;
use super::types::{ExecutionOutcomesWithReceipts, IndexerExecutionOutcomeWithReceipt};
```

**Pull request merged**